var/spack/repos/builtin/packages/c-blosc/package.py

```diff
@@ -50,6 +54,9 @@ def cmake_args(self):
             args.append('-DPREFER_EXTERNAL_ZSTD=ON')
             args.append('-DPREFER_EXTERNAL_LZ4=ON')
 
+            args.append(define('BUILD_TESTS', 'tests'))
```

Comment:
Variants should only control what is installed, not whether or not tests are run. In this case, I imagine both `c-blosc+tests` and `c-blosc~tests` result in identical installations. Instead you should use:
```python
if self.run_tests:
    args.append('-DBUILD_TESTS=ON')
else:
    args.append('-DBUILD_TESTS=OFF')
```
Then tests can be activated with `spack install --test=root c-blosc` like every other package.